### PR TITLE
Added GitHub Actions CI Workflow

### PR DESCRIPTION
**Pull request by bmiddha**
Added a GitHub Actions Workflow with a matrix strategy for Node.js 8.x, 10.x, and 12.x and across Ubuntu, MacOS, and Windows with support for node_modules caching to improve speed.

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build-node:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
```
Comment (peterblazejewicz) [inline .github/workflows/nodejs.yml]: I think v8 can be dropped, it's just reached effectily end of life as of 1st of Jan:
https://nodejs.org/en/about/releases/

Comment (bmiddha) [reply .github/workflows/nodejs.yml]: Addressed in https://github.com/microsoft/TypeScript-Node-Starter/pull/261/commits/71919c9ba183dba199f58a33e45394cce5ddc7a6

**Comment by orta** [comment]
I'm happy with this 👍 

My opinion is that it might not need to be so comprehensive (testing on Mac/window/linux) across 2 node versions? Worth bringing it down to just linux?

**Comment by peterblazejewicz** [comment]
> Worth bringing it down to just linux?
And just lastest Node LTS (12)

**Comment by orta** [comment]
haha, guess we're in alignment then :D

**Comment by bmiddha** [comment]
I wanted to add an OS build matrix to make sure the contributors get alerted by build fails if they attempt to do some OS specific commands in their code / config files. (for example using `rm` or `del` in the `scripts` section of `package.json` to clean built files.)
Should I continue to remove it?
Also, should I remove the `travis.yml` or keep both Travis CI and GitHub Actions for CI testing?
@orta @peterblazejewicz

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Get npm cache directory
+        id: npm-cache
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
+      - uses: actions/cache@v1
+        with:
+          path: ${{ steps.npm-cache.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```
Comment (peterblazejewicz) [inline .github/workflows/nodejs.yml]: Is that OK, when ending with dash?

Comment (bmiddha) [reply .github/workflows/nodejs.yml]: yes
https://github.com/actions/cache/blob/master/examples.md#using-multiple-systems-and-npm-config

Comment (peterblazejewicz) [reply .github/workflows/nodejs.yml]: ofc, sorry for missing this 🙇

**Comment by orta** [comment]
Yep 👍

**Pull request merged**